packages/ee/multi-language/lib/actions.ts

```diff
@@ -1,78 +1,90 @@
-
-  const product = await getProduct(productId);
-
-  const { hasCreateOrUpdateAccess } = await verifyUserRoleAccess(product!.organizationId, session.user?.id);
-  if (!hasCreateOrUpdateAccess) throw new AuthorizationError("Not authorized");
-
-  return await deleteLanguage(environmentId, languageId);
-};
-
-export const getSurveysUsingGivenLanguageAction = async (productId: string, languageId: string) => {
-  const session = await getServerSession(authOptions);
-  if (!session) throw new AuthorizationError("Not authorized");
-
-  const isAuthorized = await canUserAccessProduct(session.user?.id, productId);
-  if (!isAuthorized) throw new AuthorizationError("Not authorized");
-
-  return await getSurveysUsingGivenLanguage(languageId);
-};
-
-export const updateLanguageAction = async (
-  productId: string,
-  environmentId: string,
-  languageId: string,
-  languageInput: TLanguageInput
-) => {
-  const session = await getServerSession(authOptions);
-  if (!session) throw new AuthorizationError("Not authorized");
-
-  const isAuthorized = await canUserAccessProduct(session.user?.id, productId);
-  if (!isAuthorized) throw new AuthorizationError("Not authorized");
-
-  const product = await getProduct(productId);
-
-  const { hasCreateOrUpdateAccess } = await verifyUserRoleAccess(product!.organizationId, session.user?.id);
-  if (!hasCreateOrUpdateAccess) throw new AuthorizationError("Not authorized");
-
-  return await updateLanguage(environmentId, languageId, languageInput);
-};
+import {
+  getOrganizationIdFromEnvironmentId,
+  getOrganizationIdFromProductId,
+} from "@formbricks/lib/organization/utils";
+import { ZLanguageInput } from "@formbricks/types/product";
+
+const ZCreateLanguageAction = z.object({
+  productId: z.string(),
+  environmentId: z.string(),
+  languageInput: ZLanguageInput,
+});
+
+export const createLanguageAction = authenticatedActionClient
+  .schema(ZCreateLanguageAction)
+  .action(async ({ ctx, parsedInput }) => {
+    await checkAuthorization({
+      data: parsedInput.languageInput,
+      schema: ZLanguageInput,
+      userId: ctx.user.id,
+      organizationId: await getOrganizationIdFromProductId(parsedInput.productId),
+      rules: ["language", "create"],
+    });
+
+    return await createLanguage(parsedInput.productId, parsedInput.environmentId, parsedInput.languageInput);
+  });
+
+const ZDeleteLanguageAction = z.object({
+  environmentId: z.string(),
+  languageId: z.string(),
+});
+
+export const deleteLanguageAction = authenticatedActionClient
+  .schema(ZDeleteLanguageAction)
+  .action(async ({ ctx, parsedInput }) => {
+    await checkAuthorization({
+      userId: ctx.user.id,
+      organizationId: await getOrganizationIdFromEnvironmentId(parsedInput.environmentId),
+      rules: ["language", "delete"],
+    });
+
+    return await deleteLanguage(parsedInput.environmentId, parsedInput.languageId);
+  });
+
+const ZGetSurveysUsingGivenLanguageAction = z.object({
+  productId: z.string(),
```

Comment:
SECURITY RISK: We should derive the productId from the languageId, otherwise a malicious user can get the surveys for a languageId from an org where he doesn't have access by providing a productId from his org.